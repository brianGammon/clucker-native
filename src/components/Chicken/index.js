import ChickenComponent from './Chicken';

export default ChickenComponent;
