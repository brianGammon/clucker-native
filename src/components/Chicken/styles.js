import { StyleSheet } from 'react-native';
import variables from '../../styles/variables';

const iconSize = 50;
const iconPadding = 10;

export default StyleSheet.create({
  rowContainer: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'center',
  },
  iconLeft: {
    fontSize: iconSize,
    paddingLeft: iconPadding,
  },
  iconRight: {
    fontSize: iconSize,
    paddingRight: iconPadding,
  },
  moreIcon: {
    fontSize: 30,
  },
  trophyIcon: {
    fontSize: 24,
  },
  awardText: {
    marginLeft: 8,
  },
  chickenInfo: {
    marginLeft: 10,
  },
  chickenInfoText: {
    height: 20,
    fontSize: 16,
    color: variables.cluckerGrey,
  },
  fieldGroup: {
    marginBottom: 10,
  },
  row: {
    flexDirection: 'row',
  },
  flex: {
    flex: 1,
  },
  image: {
    width: 180,
    height: 180,
  },
  label: {
    fontWeight: 'bold',
  },
  h2: {
    fontWeight: 'bold',
  },
  text: {
    height: 20,
  },
  dateCell: {
    alignItems: 'center',
    padding: 5,
    width: '14.2857%',
    borderTopWidth: 1,
    borderBottomWidth: 1,
    borderLeftWidth: 1,
    borderColor: variables.cluckerGreyLight,
  },
  dateCellSuccess: {
    backgroundColor: variables.successBackground,
  },
  lastCell: {
    borderRightWidth: 1,
    borderColor: variables.cluckerGreyLight,
  },
  dateCellLabel: {
    color: variables.cluckerGrey,
    fontWeight: 'bold',
    fontSize: 16,
  },
  dateCellIcon: {
    fontSize: 20,
  },
  pastWeek: {
    paddingTop: 5,
    paddingBottom: 5,
  },
  pastWeekContainer: {
    width: '100%',
    flexDirection: 'row',
  },
});
