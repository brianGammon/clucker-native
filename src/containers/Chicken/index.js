import * as React from 'react';
import { Text, Button, View } from 'react-native';

type Props = {
  navigation: any;
}

export default class Chicken extends React.Component<Props> {
  render() {
    const { navigation } = this.props;
    const id = navigation.getParam('chickenId', 'NO-ID');
    return (
      <View style={{ flex: 1, alignItems: 'center', justifyContent: 'center' }}>
        <Text style={{ fontSize: 30 }}>{`Chicken Profile for chicken ID: ${id}`}</Text>
        <Button onPress={() => navigation.navigate('ChickenEditor')} title="Edit Chicken" />
        <Button onPress={() => navigation.goBack()} title="Go Back" />
      </View>
    );
  }
}
