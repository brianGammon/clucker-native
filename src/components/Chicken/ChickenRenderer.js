/* @flow */
import React from 'react';
import moment from 'moment';
import { Image, TouchableOpacity } from 'react-native';
import {
  Container, Content, Text, View, Button, H2, Icon,
} from 'native-base';
import ImageViewer from '../ImageViewer';
import Header from '../Header';
import Line from '../Line';
import { type Chicken, type ChickenStats } from '../../types';
import styles from './styles';

type Props = {
  navigation: any,
  prevChickenId: string,
  nextChickenId: string,
  chicken: Chicken,
  stats: ChickenStats,
  showModal: boolean,
  toggleModal: boolean => void,
  handleMoreOptions: () => void,
};

const calculateAge = (hatched: string) => {
  const hatchedAsMoment = moment(hatched);
  const now = moment();
  const months = now.diff(hatchedAsMoment, 'months');
  const years = Math.floor(months / 12);
  if (years === 0) {
    return `${months}m`;
  }
  const remainingMonths = months - years * 12;
  return `${years}y ${remainingMonths}m`;
};

const ChickenRenderer = ({
  navigation,
  chicken,
  stats,
  prevChickenId,
  nextChickenId,
  handleMoreOptions,
  showModal,
  toggleModal,
}: Props) => (
  <Container>
    <Header title="Chicken" eggButton goBackButton="Flock" />
    <Content>
      <ImageViewer
        toggleModal={toggleModal}
        showModal={showModal}
        url={chicken.photoUrl}
      />
      <View padder style={styles.rowContainer}>
        <View>
          <Button
            transparent
            dark
            disabled={prevChickenId === null}
            onPress={() => (prevChickenId
              ? navigation.replace('Chicken', { chickenId: prevChickenId })
              : null)
            }
          >
            <Icon name="arrow-dropleft-circle" />
          </Button>
        </View>

        <View style={{ flex: 1 }}>
          <H2 style={{ fontWeight: 'bold', fontSize: 20 }}>{chicken.name}</H2>
          <Text style={{ height: 20, fontSize: 16, color: 'grey' }}>
            {chicken.breed}
          </Text>
          <Text style={{ height: 20, fontSize: 16, color: 'grey' }}>
            {!!chicken.hatched
              && chicken.hatched !== ''
              && `${moment(chicken.hatched).format('MMM D, YYYY')} (${calculateAge(
                chicken.hatched,
              )})`}
          </Text>
        </View>
        <View>
          <Button
            transparent
            dark
            disabled={nextChickenId === null}
            onPress={() => (nextChickenId
              ? navigation.replace('Chicken', { chickenId: nextChickenId })
              : null)
            }
          >
            <Icon name="arrow-dropright-circle" />
          </Button>
        </View>
        <View style={styles.rowContainer}>
          <Button transparent dark onPress={handleMoreOptions}>
            <Icon active style={{ fontSize: 30 }} name="more" />
          </Button>
        </View>
      </View>
      <View padder style={{ alignItems: 'center' }}>
        <TouchableOpacity onPress={chicken.photoUrl ? () => toggleModal(!showModal) : null}>
          <Image
            style={{ width: 200, height: 200 }}
            source={
              chicken.photoUrl
                ? { uri: chicken.photoUrl }
                : require('../../assets/default-profile-photo.png')
            }
          />
        </TouchableOpacity>
      </View>

      {stats
        && stats.total > 0 && (
          <View padder>
            <Line />
            <View
              style={{
                flexDirection: 'row',
                alignItems: 'center',
                justifyContent: 'space-between',
                marginBottom: 10,
              }}
            >
              <View
                style={{
                  flex: 1,
                  alignItems: 'center',
                }}
              >
                <Text style={styles.label}>Total Eggs</Text>
                <H2 style={styles.h2}>{stats.total}</H2>
                <Text style={styles.text} />
              </View>
              {stats.heaviest && (
                <View style={{ flex: 1, alignItems: 'center' }}>
                  <Text style={[styles.label]}>Heaviest Egg</Text>
                  <H2 style={styles.h2}>
                    {stats.heaviest && stats.heaviest.weight}
                  </H2>
                  <Text style={styles.text}>
                    {stats.heaviest
                      && moment(stats.heaviest.date).format('MMM D, YYYY')}
                  </Text>
                </View>
              )}
              <View style={{ flex: 1, alignItems: 'center' }}>
                <Text style={[styles.label]}>Best Streak</Text>
                <H2 style={styles.h2}>{stats.longestStreak}</H2>
                <Text style={styles.text}>Days</Text>
              </View>
            </View>
            <Line />

            <View style={{ alignItems: 'center' }}>
              <Text style={[styles.label, { marginBottom: 10 }]}>
                Last Week
              </Text>
              <View>
                <View style={{ flexDirection: 'row' }}>
                  {Object.keys(stats.lastSevenDays || {}).map((key, index) => (
                    <View
                      key={key}
                      style={[
                        styles.dateCell,
                        stats.lastSevenDays[key] > 0
                          ? styles.dateCellSuccess
                          : null,
                        index === Object.keys(stats.lastSevenDays).length - 1
                          ? styles.lastCell
                          : null,
                      ]}
                    >
                      <Text style={styles.dateCellLabel}>
                        {moment(key).format('MMM')}
                      </Text>
                      <Text style={styles.dateCellLabel}>
                        {moment(key).format('D')}
                      </Text>
                      <View>
                        <Icon
                          style={styles.dateCellIcon}
                          name={
                            stats.lastSevenDays[key] > 0
                              ? 'checkmark-circle'
                              : 'close'
                          }
                        />
                      </View>
                    </View>
                  ))}
                </View>
              </View>
            </View>
          </View>
      )}
    </Content>
  </Container>
);

export default ChickenRenderer;
