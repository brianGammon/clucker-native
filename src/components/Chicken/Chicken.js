/* @flow */
import * as React from 'react';
import { connect } from 'react-redux';
import { ActionSheet } from 'native-base';
import { Text, Alert } from 'react-native';
import ChickenRenderer from './ChickenRenderer';
import chickenSelector from '../../selectors/chickenSelector';
import chickenStatsSelector from '../../selectors/chickenStatsSelector';
import { actionTypes } from '../../redux/constants';
import {
  type Chicken as ChickenType,
  type Navigation,
  type ChickenStats,
} from '../../types';

type Props = {
  navigation: Navigation,
  chicken: ChickenType,
  stats: ChickenStats,
  chickenId: string,
  prevChickenId: string,
  nextChickenId: string,
  flockId: string,
  deleteChicken: (flockId: string, chickenId: string, paths: string[]) => void,
};

type State = {
  showModal: boolean,
};

class Chicken extends React.Component<Props, State> {
  state = {
    showModal: false,
  };

  shouldComponentUpdate(newProps) {
    const { chicken, navigation } = newProps;
    if (!chicken) {
      navigation.goBack();
      return false;
    }
    return true;
  }

  toggleModal = (visible) => {
    this.setState({ showModal: visible });
  };

  handleMoreOptions = () => {
    const { navigation, chickenId } = this.props;
    const BUTTONS = ['Edit Chicken', 'Delete Chicken', 'Cancel'];
    const ACTIONS = [
      () => navigation.navigate('ChickenEditor', { chickenId }),
      () => this.deleteChicken(chickenId),
      () => {},
    ];
    const DESTRUCTIVE_INDEX = 1;
    const CANCEL_INDEX = 2;
    ActionSheet.show(
      {
        options: BUTTONS,
        cancelButtonIndex: CANCEL_INDEX,
        destructiveButtonIndex: DESTRUCTIVE_INDEX,
        title: 'Additonal Actions',
      },
      (buttonIndex) => {
        // console.log(BUTTONS[buttonIndex]);
        ACTIONS[buttonIndex]();
      },
    );
  };

  deleteChicken = (chickenId) => {
    const { flockId, deleteChicken, chicken } = this.props;

    const paths = [];
    if (chicken.photoPath && chicken.photoPath !== '') {
      paths.push(chicken.photoPath);
    }
    if (chicken.thumbnailPath && chicken.thumbnailPath !== '') {
      paths.push(chicken.thumbnailPath);
    }

    Alert.alert(
      'Are you sure?',
      `This will delete ${chicken.name} and all her eggs.`,
      [
        {
          text: 'Cancel',
        },
        {
          text: 'Delete',
          onPress: () => deleteChicken(flockId, chickenId, paths),
          style: 'destructive',
        },
      ],
    );
  };

  render() {
    const {
      chicken,
      chickenId,
      prevChickenId,
      nextChickenId,
      stats,
      navigation,
    } = this.props;

    const { showModal } = this.state;

    if (chickenId === 'NO-ID') {
      return <Text>No Chicken ID passed in!</Text>;
    }

    if (!chicken) {
      return <Text>{`Chicken with ID ${chickenId} not found`}</Text>;
    }

    return (
      <ChickenRenderer
        navigation={navigation}
        chicken={chicken}
        chickenId={chickenId}
        stats={stats}
        prevChickenId={prevChickenId}
        nextChickenId={nextChickenId}
        showModal={showModal}
        toggleModal={this.toggleModal}
        handleMoreOptions={this.handleMoreOptions}
      />
    );
  }
}

const mapStateToProps = ({ chickens, eggs, userSettings }, { navigation }) => {
  const chickenId = navigation.getParam('chickenId', 'NO-ID');
  return {
    ...chickenSelector(chickens.data, chickenId),
    stats: chickenStatsSelector(eggs.data, chickenId),
    flockId: userSettings.data.currentFlockId,
  };
};

const mapDispatchtoProps = dispatch => ({
  deleteChicken: (flockId, chickenId, paths) => dispatch({
    type: actionTypes.DELETE_CHICKEN_REQUESTED,
    payload: { flockId, chickenId, paths },
  }),
});

export default connect(
  mapStateToProps,
  mapDispatchtoProps,
)(Chicken);
