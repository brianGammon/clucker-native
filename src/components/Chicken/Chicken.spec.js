import React from 'react';
import configureStore from 'redux-mock-store';
import Chicken from '.';

describe('Chicken component:', () => {
  let paramReturned = 'chicken1';
  const middlewares = [];
  const mockStore = configureStore(middlewares);
  const store = mockStore({
    chickens: {
      items: {
        chicken1: {
          name: 'The First Chicken',
        },
        chicken2: {
          name: 'The Second Chicken',
        },
      },
    },
  });

  const navigation = {
    getParam: jest.fn((param, defaultId) => {
      if (paramReturned) {
        return paramReturned;
      }
      return defaultId;
    }),
  };

  test('Should render correctly when chicken ID passed in', () => {
    paramReturned = 'chicken1';
    const wrapper = shallow(
      <Chicken store={store} navigation={navigation} />,
    ).dive();
    expect(wrapper).toMatchSnapshot();
  });

  test('Should render correctly when no chicken ID passed in', () => {
    paramReturned = null;
    const wrapper = shallow(
      <Chicken store={store} navigation={navigation} />,
    ).dive();
    expect(wrapper).toMatchSnapshot();
  });

  test('Should render correctly when chicken not found', () => {
    paramReturned = 'chicken3';
    const wrapper = shallow(
      <Chicken store={store} navigation={navigation} />,
    ).dive();
    expect(wrapper).toMatchSnapshot();
  });
});
