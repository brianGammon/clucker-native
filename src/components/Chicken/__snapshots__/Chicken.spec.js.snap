// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Chicken component: ChickenRenderer test ChickenRenderer should render correctly when no stats yet 1`] = `
<Styled(Container)>
  <withNavigation(Header)
    eggButton={true}
    goBackButton="Flock"
    title="Chicken"
  />
  <Styled(Content)>
    <ImageViewer
      showModal={true}
      toggleModal={[Function]}
      url="https://firebasestorage.googleapis.com/v0/b/cluckr-prod.appspot.com/o/uploads%2Fuser%3AvEFMHqRcozMSriuqeJNKlCXnGMr2%2Fflock%3Aflock1%2F1BD434AA-4A5D-40A9-A97A-C3D64FC8B841.jpeg?alt=media&token=029f7593-9ed3-481e-8dc6-5d3c38844180"
    />
    <Styled(ViewNB)
      padder={true}
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-between",
        }
      }
    >
      <Styled(ViewNB)>
        <Styled(Button)
          dark={true}
          disabled={true}
          onPress={[Function]}
          transparent={true}
        >
          <Styled(Icon)
            name="arrow-dropleft"
          />
        </Styled(Button)>
      </Styled(ViewNB)>
      <Styled(ViewNB)
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <Styled(H2)
          style={
            Object {
              "fontSize": 20,
              "fontWeight": "bold",
            }
          }
        >
          Chicken1
        </Styled(H2)>
        <Styled(Text)
          style={
            Object {
              "color": "grey",
              "fontSize": 16,
              "height": 20,
            }
          }
        >
          Cuckoo Maran
        </Styled(Text)>
        <Styled(Text)
          style={
            Object {
              "color": "grey",
              "fontSize": 16,
              "height": 20,
            }
          }
        >
          Oct 5, 2016 (1y 11m)
        </Styled(Text)>
      </Styled(ViewNB)>
      <Styled(ViewNB)
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "space-between",
          }
        }
      >
        <Styled(Button)
          dark={true}
          transparent={true}
        >
          <Styled(Icon)
            active={true}
            name="more"
            style={
              Object {
                "fontSize": 30,
              }
            }
          />
        </Styled(Button)>
      </Styled(ViewNB)>
      <Styled(ViewNB)>
        <Styled(Button)
          dark={true}
          disabled={false}
          onPress={[Function]}
          transparent={true}
        >
          <Styled(Icon)
            name="arrow-dropright"
          />
        </Styled(Button)>
      </Styled(ViewNB)>
    </Styled(ViewNB)>
    <Styled(ViewNB)
      padder={true}
      style={
        Object {
          "alignItems": "center",
        }
      }
    >
      <TouchableOpacity
        activeOpacity={0.2}
        onPress={[Function]}
      >
        <Image
          source={
            Object {
              "uri": "https://firebasestorage.googleapis.com/v0/b/cluckr-prod.appspot.com/o/uploads%2Fuser%3AvEFMHqRcozMSriuqeJNKlCXnGMr2%2Fflock%3Aflock1%2F1BD434AA-4A5D-40A9-A97A-C3D64FC8B841.jpeg?alt=media&token=029f7593-9ed3-481e-8dc6-5d3c38844180",
            }
          }
          style={
            Object {
              "height": 200,
              "width": 200,
            }
          }
        />
      </TouchableOpacity>
    </Styled(ViewNB)>
  </Styled(Content)>
</Styled(Container)>
`;

exports[`Chicken component: ChickenRenderer test ChickenRenderer should render correctly when no weight yet 1`] = `
<Styled(Container)>
  <withNavigation(Header)
    eggButton={true}
    goBackButton="Flock"
    title="Chicken"
  />
  <Styled(Content)>
    <ImageViewer
      showModal={true}
      toggleModal={[Function]}
      url="https://firebasestorage.googleapis.com/v0/b/cluckr-prod.appspot.com/o/uploads%2Fuser%3AvEFMHqRcozMSriuqeJNKlCXnGMr2%2Fflock%3Aflock1%2F1BD434AA-4A5D-40A9-A97A-C3D64FC8B841.jpeg?alt=media&token=029f7593-9ed3-481e-8dc6-5d3c38844180"
    />
    <Styled(ViewNB)
      padder={true}
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-between",
        }
      }
    >
      <Styled(ViewNB)>
        <Styled(Button)
          dark={true}
          disabled={true}
          onPress={[Function]}
          transparent={true}
        >
          <Styled(Icon)
            name="arrow-dropleft"
          />
        </Styled(Button)>
      </Styled(ViewNB)>
      <Styled(ViewNB)
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <Styled(H2)
          style={
            Object {
              "fontSize": 20,
              "fontWeight": "bold",
            }
          }
        >
          Chicken1
        </Styled(H2)>
        <Styled(Text)
          style={
            Object {
              "color": "grey",
              "fontSize": 16,
              "height": 20,
            }
          }
        >
          Cuckoo Maran
        </Styled(Text)>
        <Styled(Text)
          style={
            Object {
              "color": "grey",
              "fontSize": 16,
              "height": 20,
            }
          }
        >
          Oct 5, 2016 (1y 11m)
        </Styled(Text)>
      </Styled(ViewNB)>
      <Styled(ViewNB)
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "space-between",
          }
        }
      >
        <Styled(Button)
          dark={true}
          transparent={true}
        >
          <Styled(Icon)
            active={true}
            name="more"
            style={
              Object {
                "fontSize": 30,
              }
            }
          />
        </Styled(Button)>
      </Styled(ViewNB)>
      <Styled(ViewNB)>
        <Styled(Button)
          dark={true}
          disabled={false}
          onPress={[Function]}
          transparent={true}
        >
          <Styled(Icon)
            name="arrow-dropright"
          />
        </Styled(Button)>
      </Styled(ViewNB)>
    </Styled(ViewNB)>
    <Styled(ViewNB)
      padder={true}
      style={
        Object {
          "alignItems": "center",
        }
      }
    >
      <TouchableOpacity
        activeOpacity={0.2}
        onPress={[Function]}
      >
        <Image
          source={
            Object {
              "uri": "https://firebasestorage.googleapis.com/v0/b/cluckr-prod.appspot.com/o/uploads%2Fuser%3AvEFMHqRcozMSriuqeJNKlCXnGMr2%2Fflock%3Aflock1%2F1BD434AA-4A5D-40A9-A97A-C3D64FC8B841.jpeg?alt=media&token=029f7593-9ed3-481e-8dc6-5d3c38844180",
            }
          }
          style={
            Object {
              "height": 200,
              "width": 200,
            }
          }
        />
      </TouchableOpacity>
    </Styled(ViewNB)>
    <Styled(ViewNB)
      padder={true}
    >
      <Line />
      <Styled(ViewNB)
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "space-between",
            "marginBottom": 10,
          }
        }
      >
        <Styled(ViewNB)
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
            }
          }
        >
          <Styled(Text)
            style={
              Object {
                "color": "grey",
                "fontSize": 17,
                "fontWeight": "bold",
              }
            }
          >
            Total Eggs
          </Styled(Text)>
          <Styled(H2)
            style={
              Object {
                "fontWeight": "bold",
              }
            }
          >
            1
          </Styled(H2)>
          <Styled(Text)
            style={
              Object {
                "height": 20,
              }
            }
          />
        </Styled(ViewNB)>
        <Styled(ViewNB)
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
            }
          }
        >
          <Styled(Text)
            style={
              Array [
                Object {
                  "color": "grey",
                  "fontSize": 17,
                  "fontWeight": "bold",
                },
              ]
            }
          >
            Best Streak
          </Styled(Text)>
          <Styled(H2)
            style={
              Object {
                "fontWeight": "bold",
              }
            }
          >
            0
          </Styled(H2)>
          <Styled(Text)
            style={
              Object {
                "height": 20,
              }
            }
          >
            Days
          </Styled(Text)>
        </Styled(ViewNB)>
      </Styled(ViewNB)>
      <Line />
      <Styled(ViewNB)
        style={
          Object {
            "alignItems": "center",
          }
        }
      >
        <Styled(Text)
          style={
            Array [
              Object {
                "color": "grey",
                "fontSize": 17,
                "fontWeight": "bold",
              },
              Object {
                "marginBottom": 10,
              },
            ]
          }
        >
          Last Week
        </Styled(Text)>
        <Styled(ViewNB)
          style={Object {}}
        >
          <Styled(ViewNB)
            style={
              Object {
                "flexDirection": "row",
              }
            }
          >
            <Styled(ViewNB)
              key="2018-10-06"
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "borderBottomWidth": 1,
                    "borderColor": "grey",
                    "borderLeftWidth": 1,
                    "borderTopWidth": 1,
                    "padding": 10,
                  },
                  Object {
                    "backgroundColor": "lightgreen",
                  },
                  Object {
                    "borderColor": "grey",
                    "borderRightWidth": 1,
                  },
                ]
              }
            >
              <Styled(Text)
                style={
                  Object {
                    "color": "grey",
                    "fontSize": 16,
                    "fontWeight": "bold",
                  }
                }
              >
                Oct
              </Styled(Text)>
              <Styled(Text)
                style={
                  Object {
                    "color": "grey",
                    "fontSize": 16,
                    "fontWeight": "bold",
                  }
                }
              >
                6
              </Styled(Text)>
              <Styled(ViewNB)>
                <Styled(Icon)
                  name="checkmark-circle"
                  style={
                    Object {
                      "fontSize": 20,
                    }
                  }
                />
              </Styled(ViewNB)>
            </Styled(ViewNB)>
          </Styled(ViewNB)>
        </Styled(ViewNB)>
      </Styled(ViewNB)>
    </Styled(ViewNB)>
  </Styled(Content)>
</Styled(Container)>
`;

exports[`Chicken component: ChickenRenderer test ChickenRenderer should render correctly with all stats available 1`] = `
<Styled(Container)>
  <withNavigation(Header)
    eggButton={true}
    goBackButton="Flock"
    title="Chicken"
  />
  <Styled(Content)>
    <ImageViewer
      showModal={true}
      toggleModal={[Function]}
      url="https://firebasestorage.googleapis.com/v0/b/cluckr-prod.appspot.com/o/uploads%2Fuser%3AvEFMHqRcozMSriuqeJNKlCXnGMr2%2Fflock%3Aflock1%2F1BD434AA-4A5D-40A9-A97A-C3D64FC8B841.jpeg?alt=media&token=029f7593-9ed3-481e-8dc6-5d3c38844180"
    />
    <Styled(ViewNB)
      padder={true}
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-between",
        }
      }
    >
      <Styled(ViewNB)>
        <Styled(Button)
          dark={true}
          disabled={true}
          onPress={[Function]}
          transparent={true}
        >
          <Styled(Icon)
            name="arrow-dropleft"
          />
        </Styled(Button)>
      </Styled(ViewNB)>
      <Styled(ViewNB)
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <Styled(H2)
          style={
            Object {
              "fontSize": 20,
              "fontWeight": "bold",
            }
          }
        >
          Chicken1
        </Styled(H2)>
        <Styled(Text)
          style={
            Object {
              "color": "grey",
              "fontSize": 16,
              "height": 20,
            }
          }
        >
          Cuckoo Maran
        </Styled(Text)>
        <Styled(Text)
          style={
            Object {
              "color": "grey",
              "fontSize": 16,
              "height": 20,
            }
          }
        >
          Oct 5, 2016 (1y 11m)
        </Styled(Text)>
      </Styled(ViewNB)>
      <Styled(ViewNB)
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "space-between",
          }
        }
      >
        <Styled(Button)
          dark={true}
          transparent={true}
        >
          <Styled(Icon)
            active={true}
            name="more"
            style={
              Object {
                "fontSize": 30,
              }
            }
          />
        </Styled(Button)>
      </Styled(ViewNB)>
      <Styled(ViewNB)>
        <Styled(Button)
          dark={true}
          disabled={false}
          onPress={[Function]}
          transparent={true}
        >
          <Styled(Icon)
            name="arrow-dropright"
          />
        </Styled(Button)>
      </Styled(ViewNB)>
    </Styled(ViewNB)>
    <Styled(ViewNB)
      padder={true}
      style={
        Object {
          "alignItems": "center",
        }
      }
    >
      <TouchableOpacity
        activeOpacity={0.2}
        onPress={[Function]}
      >
        <Image
          source={
            Object {
              "uri": "https://firebasestorage.googleapis.com/v0/b/cluckr-prod.appspot.com/o/uploads%2Fuser%3AvEFMHqRcozMSriuqeJNKlCXnGMr2%2Fflock%3Aflock1%2F1BD434AA-4A5D-40A9-A97A-C3D64FC8B841.jpeg?alt=media&token=029f7593-9ed3-481e-8dc6-5d3c38844180",
            }
          }
          style={
            Object {
              "height": 200,
              "width": 200,
            }
          }
        />
      </TouchableOpacity>
    </Styled(ViewNB)>
    <Styled(ViewNB)
      padder={true}
    >
      <Line />
      <Styled(ViewNB)
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "space-between",
            "marginBottom": 10,
          }
        }
      >
        <Styled(ViewNB)
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
            }
          }
        >
          <Styled(Text)
            style={
              Object {
                "color": "grey",
                "fontSize": 17,
                "fontWeight": "bold",
              }
            }
          >
            Total Eggs
          </Styled(Text)>
          <Styled(H2)
            style={
              Object {
                "fontWeight": "bold",
              }
            }
          >
            1
          </Styled(H2)>
          <Styled(Text)
            style={
              Object {
                "height": 20,
              }
            }
          />
        </Styled(ViewNB)>
        <Styled(ViewNB)
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
            }
          }
        >
          <Styled(Text)
            style={
              Array [
                Object {
                  "color": "grey",
                  "fontSize": 17,
                  "fontWeight": "bold",
                },
              ]
            }
          >
            Heaviest Egg
          </Styled(Text)>
          <Styled(H2)
            style={
              Object {
                "fontWeight": "bold",
              }
            }
          >
            88.1
          </Styled(H2)>
          <Styled(Text)
            style={
              Object {
                "height": 20,
              }
            }
          >
            Apr 2, 2017
          </Styled(Text)>
        </Styled(ViewNB)>
        <Styled(ViewNB)
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
            }
          }
        >
          <Styled(Text)
            style={
              Array [
                Object {
                  "color": "grey",
                  "fontSize": 17,
                  "fontWeight": "bold",
                },
              ]
            }
          >
            Best Streak
          </Styled(Text)>
          <Styled(H2)
            style={
              Object {
                "fontWeight": "bold",
              }
            }
          >
            0
          </Styled(H2)>
          <Styled(Text)
            style={
              Object {
                "height": 20,
              }
            }
          >
            Days
          </Styled(Text)>
        </Styled(ViewNB)>
      </Styled(ViewNB)>
      <Line />
      <Styled(ViewNB)
        style={
          Object {
            "alignItems": "center",
          }
        }
      >
        <Styled(Text)
          style={
            Array [
              Object {
                "color": "grey",
                "fontSize": 17,
                "fontWeight": "bold",
              },
              Object {
                "marginBottom": 10,
              },
            ]
          }
        >
          Last Week
        </Styled(Text)>
        <Styled(ViewNB)
          style={Object {}}
        >
          <Styled(ViewNB)
            style={
              Object {
                "flexDirection": "row",
              }
            }
          >
            <Styled(ViewNB)
              key="2018-10-06"
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "borderBottomWidth": 1,
                    "borderColor": "grey",
                    "borderLeftWidth": 1,
                    "borderTopWidth": 1,
                    "padding": 10,
                  },
                  Object {
                    "backgroundColor": "lightgreen",
                  },
                  Object {
                    "borderColor": "grey",
                    "borderRightWidth": 1,
                  },
                ]
              }
            >
              <Styled(Text)
                style={
                  Object {
                    "color": "grey",
                    "fontSize": 16,
                    "fontWeight": "bold",
                  }
                }
              >
                Oct
              </Styled(Text)>
              <Styled(Text)
                style={
                  Object {
                    "color": "grey",
                    "fontSize": 16,
                    "fontWeight": "bold",
                  }
                }
              >
                6
              </Styled(Text)>
              <Styled(ViewNB)>
                <Styled(Icon)
                  name="checkmark-circle"
                  style={
                    Object {
                      "fontSize": 20,
                    }
                  }
                />
              </Styled(ViewNB)>
            </Styled(ViewNB)>
          </Styled(ViewNB)>
        </Styled(ViewNB)>
      </Styled(ViewNB)>
    </Styled(ViewNB)>
  </Styled(Content)>
</Styled(Container)>
`;

exports[`Chicken component: Should render correctly when chicken ID passed in 1`] = `
<ChickenRenderer
  chicken={
    Object {
      "breed": "Cuckoo Maran",
      "hatched": "2016-10-05",
      "name": "Chicken1",
      "photoPath": "uploads/user:user1/flock:flock1/1BD434AA-4A5D-40A9-A97A-C3D64FC8B841.jpeg",
      "photoUrl": "https://firebasestorage.googleapis.com/v0/b/cluckr-prod.appspot.com/o/uploads%2Fuser%3AvEFMHqRcozMSriuqeJNKlCXnGMr2%2Fflock%3Aflock1%2F1BD434AA-4A5D-40A9-A97A-C3D64FC8B841.jpeg?alt=media&token=029f7593-9ed3-481e-8dc6-5d3c38844180",
      "thumbnailPath": "uploads/user:user1/flock:flock1/thumb_1BD434AA-4A5D-40A9-A97A-C3D64FC8B841.jpeg",
      "thumbnailUrl": "https://firebasestorage.googleapis.com/v0/b/cluckr-prod.appspot.com/o/uploads%2Fuser%3AvEFMHqRcozMSriuqeJNKlCXnGMr2%2Fflock%3Aflock1%2Fthumb_1BD434AA-4A5D-40A9-A97A-C3D64FC8B841.jpeg?alt=media&token=689dfae8-c034-46ae-9984-db5c4ee3fe72",
    }
  }
  chickenId="chicken1"
  handleMoreOptions={[Function]}
  navigation={
    Object {
      "getParam": [MockFunction] {
        "calls": Array [
          Array [
            "chickenId",
            "NO-ID",
          ],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": "chicken1",
          },
        ],
      },
    }
  }
  nextChickenId="chicken2"
  prevChickenId={null}
  showModal={false}
  stats={
    Object {
      "heaviest": Object {
        "chickenId": "chicken1",
        "chickenName": "Chicken1",
        "date": "2018-02-13",
        "modified": "2018-02-13T13:45:46.640Z",
        "userId": "user1",
        "weight": "73.5",
      },
      "lastSevenDays": Object {
        "2018-05-09": 1,
        "2018-05-10": 0,
        "2018-05-11": 1,
        "2018-05-12": 1,
        "2018-05-13": 1,
        "2018-05-14": 0,
        "2018-05-15": 0,
      },
      "longestStreak": 10,
      "total": 78,
    }
  }
  toggleModal={[Function]}
/>
`;

exports[`Chicken component: Should render correctly when chicken not found 1`] = `
<Styled(Text)>
  Chicken with ID chicken5 not found
</Styled(Text)>
`;

exports[`Chicken component: Should render correctly when no chicken ID passed in 1`] = `
<Styled(Text)>
  No Chicken ID passed in!
</Styled(Text)>
`;
